frcxx_nft/src/state.rs

```diff
@@ -234,47 +215,56 @@ impl NFTState {
 
     /// Approves an operator to transfer a set of specified tokens
     ///
-    /// Checks that the caller is the owner of the specified token. If any of the token_ids is not
-    /// valid (i.e. non-existent or not-owned by the caller), the entire batch approval is aborted.. If any of the token_ids is not
-    /// valid (i.e. non-existent or not-owned by the caller), the entire batch approval is aborted.
-    pub fn approve_for_tokens<BS: Blockstore>(
+    /// The caller should own the tokens or an account-level operator on the owner of the tokens.
```

Comment:
I think all of these "the caller" comments are out of place here. This is a pure data object, the caller is some library code. Describe what it does, don't prescribe how to use it.